### ISSUE TITLE
could we revert libxml2 to support both python2 and 3 ?

### ISSUE THREAD

**Issue by revelator** [msys2/MINGW-packages#6402]
Some packages like gnome-doc-utils still need it, we can skip the numpy dependency without any problems as far as i have been able to determine (atleast for gnome-doc-utils).
fedora has released a patch updating gnome-doc-utils for python3 support, unfortunatly it seems its not working correctly on windows (utf8 error).

**Comment by Alexpux** [comment]
Where gnome-doc-utils used? As i know it not used for a long time. Can you point me to patch from fedora?

**Comment by revelator** [comment]
patch hosted here [](https://github.com/Oleh-Kravchenko/portage/tree/master/app-text/gnome-doc-utils/files)

still used in clutter clutter-gst cogl libgda and pitivi according to PKGBUILD files.

**Comment by lazka** [comment]
My plan was to drop gnome-doc-utils next (by either patching things or dropping packages, since cogl and clutter are not really maintained and on the way out)

**Comment by lazka** [comment]
@revelator What's the problem you are facing with gnome-doc-utils not working?

**Comment by revelator** [comment]
Ah ok. Well the problem with the fedora python3 patches for gnome-doc-utils (i suspect you mean those ?) are that the build fails with some utf8 errors. Still investigating causes but it seems related to our python3 package not handling utf8 well.

**Comment by lazka** [comment]
I mean what's the actual problem that motivated you opening this issue :)

**Comment by revelator** [comment]
i was updating my private fork after the deprecation of most 
python2 packages and noticed gnome-doc-utils was no longer building because of it.

I maintain a special build of your mingw-w64-gcc with the now defunct TDM patchset (which i have updated continuously) for codeblocks users who prefer static runtime linking while still being able to throw exceptions across dll boundaries.

I try to make sure as many of your packages still build with this fork to avoid problems that earlier versions of TDMs gcc compilers had. Just had the notion that you might appreciate getting bug reports and potential fixes.

**Comment by revelator** [comment]
clutter is still needed by eye of gnome plugins i think ?, and cogl seems to be used by clutter, pitivi does not seem to be used by any package anymore so that might go.

Not sure how much relevance clutter has in eog or if it can be replaced by something different.

**Comment by lazka** [comment]
ok, thanks for the context. it's appreciated :)

**Comment by revelator** [comment]
No problem :+1: 

If there is any interrest i can provide the updated TDM patchset for those wanting to do specialized gcc builds. It was recently upated for gcc-9.3.0 as well and works like a charm in most cases.

It will however still link to the libstdc++ dll if building some c++ code unless you rename the import library.

I havent found a foolprof way to get around this yet with code changes, but if you build a static gcc it works without this hack. 

I also removed TDMs specialized winpthreads build so threaded code will allways depend on the winpthread dll. 

The reason this was done was because it broke several packages that expect the shared version, it could probably be patched out of those but it would be a lot of work.

**Comment by lazka** [comment]
Consider opening a new issue for this, so others see it too (I don't know much about the toolchain...)

**Comment by revelator** [comment]
Will do.

**Comment by Ede123** [comment]
FWIW we're still using gnome-doc-utils (specifically xml2po) to create documentation for Inkscape.

The documentation is built separately (so it does not affect the Inkscape build itself) but we still depend on xml2po until somebody finds the time to migrate to something more modern.

**Comment by lazka** [comment]
Here is the fedora patch: https://src.fedoraproject.org/rpms/gnome-doc-utils/blob/master/f/gnome-doc-utils-0.20.10-python3.patch

I'll give that a try.
